README.md

```diff
@@ -1,2 +1,122 @@
-# graphql-directive-uid
-@uid directive for GraphQL
+<div align="center">
+  <h1>@uid directive for GraphQL</h1>
+  <p>Directive for GraphQL schema.</p>
+</div>
+
+<hr />
+
+## The problem
+
+A lot of people have problem with generating unique `ID` for each graphql schema type
+
+## This solution
+
+The `@uid` is the simple graphql directive that allo user to have minimum effort but have all benefits from graphql cache.
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+* [Installation](#installation)
+* [Usage](#usage)
+* [More on `data-testid`s](#more-on-data-testids)
+* [Examples](#examples)
+* [Contributors](#contributors)
+* [LICENSE](#license)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Installation
+
+This module is distributed via [npm][npm] which is bundled with [node][node] and
+should be installed as one of your project's `devDependencies`:
+
+use npm:
+
+```
+npm install --save graphql-directive-uid
+```
+
+or yarn :
+
+```
+yarn add graphql-directive-uid
+```
+
+This library has a `peerDependencies` listing for:
+
+* `graphql`
+* `graphql-tools`
+
+## Usage
+
+```javascript
+import { makeExecutableSchema } from 'graphql-tools';
+import graphqlDirectiveUid from 'graphql-directive-uid';
+import typeDefs from './schema.graphql';
+import resolvers from './resolvers';
+
+export default makeExecutableSchema({
+  typeDefs,
+  resolvers,
+  schemaDirectives: {
+    uid: graphqlDirectiveUid, // the name of directive can be as you like
+  },
+});
+```
+
+GraphQL schema:
+
+```graphql
+type Person @uid(from: ["personId", "name"]) {
+  personId: Int
+  name: String
+}
+```
+
+## Examples
```

Comment:
A sample query and its output (from the example) would be nice to have inlined here